apps/src/templates/codeDocs/EmbeddedBlock.jsx

```diff
@@ -26,7 +22,7 @@ export default function EmbeddedBlock({blockName, link, ariaLabel}) {
       shrinkBlockSpaceContainer(blockSpace, true);
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
(it being ignoring the exhaustive dependencies rule)

Reply:
If we remove the ignore rule it complains about `blockRef` being missing. (It expects both `blockName` and `blockRef`.) The rule was added when we updated the linter: https://github.com/code-dot-org/code-dot-org/pull/51059

I checked with @bethanyaconnor who originally built the feature. She said she ran into the duplicate block issue with CDO Blockly too but that adjusting the `useEffect` dependencies at the time fixed it. FWIW, we don't see duplicate blocks with CDO Blockly today, but it does look like we're infinitely creating read-only blockspaces inside the same embedded code div.

Reply:
Found a fix with @bethanyaconnor! https://github.com/code-dot-org/code-dot-org/pull/59031/commits/5e9c54c36d12133d9b87d47fc30e36707fce3bf0

Since this component gets remounted multiple times as part of `LessonOverview`, we want to make sure we're keeping the same reference. Using `useRef` instead of `createRef` is typically the way to do that in functional components. This way, `blockRef` doesn't change so it can included as a dependency. The latest commit removes the eslint disable.